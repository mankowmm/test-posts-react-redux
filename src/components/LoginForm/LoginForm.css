.LoginForm {
    margin: 30px;
}