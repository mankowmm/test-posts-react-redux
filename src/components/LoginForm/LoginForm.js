import React from 'react';
import { Loader } from '../Loader/Loader'
import { Button, Form, FormGroup, Label, Input, FormText, Alert } from 'reactstrap';
import { ErrorMessage } from '../ErrorMessage/ErrorMessage'
import { AuthHelper ,PASSWORD_VALIDATION_MESSAGE, USERNAME_VALIDATION_MESSAGE } from '../../helpers/authHelper';
import './LoginForm.css';

export const LoginForm = ({authenticatedUserName, authenticating, errorLogin, errorUserName, errorPassword, authenticateUser }) => {

    console.log('error', errorLogin, errorUserName, errorPassword);

    

    const renderForm = ()=> {
        
   
        return (
            <Form onSubmit={(e) => {
                    const userNameVal = e.target.username.value;
                    const passwordVal = e.target.password.value;
                    authenticateUser(userNameVal, passwordVal);
                    e.preventDefault();
                }}>
                <FormGroup>
                    <Label for="username">Username</Label>
                    <Input type="text" name="username" id="username" placeholder="User Name" required />
                    <div id="usernamevalidation"></div>
                </FormGroup>
                <FormGroup>
                    <Label for="password">Password</Label>
                    <Input type="password" name="password" id="password" placeholder="Password" required />
                    <div id="passwordvalidation"></div>
                </FormGroup>
                <Button>Submit</Button>
            </Form> 
               
        )
    }
    
    let content = renderForm();
    const errorMessageUserName = errorUserName ? (
        <Alert color="danger">
            {errorUserName.message}
        </Alert>
      ) : null;

    const errorMessagePassword = errorPassword ? (
    <Alert color="danger">
        {errorPassword.message}
    </Alert>
    ) : null;  
    
    return (
        <div className="LoginForm">{content}<br/>{errorMessageUserName}{errorMessagePassword}</div>
    )
}