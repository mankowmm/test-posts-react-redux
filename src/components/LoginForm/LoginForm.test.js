import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { LoginForm } from './LoginForm';

describe('LoginForm component test suite', ()=> {

    let elWrapper;
    beforeEach(() => {
        
    })

    it('should validate with snapshot', () => {
        const tree = renderer
          .create(<LoginForm/>)
          .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should render form with inputs and button', ()=> {
        elWrapper = mount(<LoginForm/>);
        expect(elWrapper.find('form').length).toBe(1);
        expect(elWrapper.find('input').length).toBe(2);
        expect(elWrapper.find('button').length).toBe(1);
    })
   

});