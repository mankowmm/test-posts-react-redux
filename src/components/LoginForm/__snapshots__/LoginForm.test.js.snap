// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm component test suite should validate with snapshot 1`] = `
<div
  className="LoginForm"
>
  <form
    className=""
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label
        className=""
        htmlFor="username"
      >
        Username
      </label>
      <input
        className="form-control"
        id="username"
        name="username"
        placeholder="User Name"
        required={true}
        type="text"
      />
      <div
        id="usernamevalidation"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className=""
        htmlFor="password"
      >
        Password
      </label>
      <input
        className="form-control"
        id="password"
        name="password"
        placeholder="Password"
        required={true}
        type="password"
      />
      <div
        id="passwordvalidation"
      />
    </div>
    <button
      className="btn btn-secondary"
      onClick={[Function]}
      type={undefined}
    >
      Submit
    </button>
  </form>
  <br />
</div>
`;
